metricsbp/statsd.go

```diff
@@ -354,6 +331,9 @@ func (st *Statsd) Close() error {
 // so in most cases you shouldn't be using it in production code.
 // But it's useful in unit tests to verify that you have the correct metrics you
 // want to report.
+//
+// When you use this in test code you also want to set BufferInMemoryForTesting
+// to true in the statsd Config, otherwise your test could become flaky.
```

Comment:
Wouldn't it just fail? since it's dropping data, why would it sometimes work and sometimes fail?

Reply:
the dropping of the data happens on a ticker. it's possible/likely when the test runs the ticker hasn't fired yet.